build.lua

```diff
@@ -47,51 +47,47 @@ maxprintline = 9999
 ctanzip = "pgf.ctan.flatdir"
 packtdszip = true
 
--- For the way pgf does releases
-local function trim(str)
-    return str:gsub("^%s*(.-)%s$", "%1")
+-- if ctanupload is nil or false, only validation is attempted
+if options["dry-run"] then
+  ctanupload = false
 end
 
-local function runcmd(cmd)
-    local pid = assert(io.popen(cmd))
-    local out = trim(pid:read("*all"))
-    pid:close()
-    return out
-end
+-- CTAN upload
+uploadconfig = {
+  announcement_file = "doc/generic/pgf/RELEASE_NOTES.md",
+  author = "Christian Feuersänger;Henri Menke;The PGF/TikZ Team;Till Tantau",
+  bugs = "https://github.com/pgf-tikz/pgf/issues",
+  ctanPath = "/graphics/pgf/base",
+  description = [[<p>PGF is a macro package for creating graphics. It is platform- and format-independent and works together with the most important TeX backend drivers, including pdfTeX and dvips. It comes with a user-friendly syntax layer called TikZ.<br></p><p>Its usage is similar to <a data-cke-saved-href="/pkg/pstricks-base" href="/pkg/pstricks-base">pstricks</a> and the standard picture environment. PGF works with plain (pdf-)TeX, (pdf-)LaTeX, and ConTeXt. Unlike <a data-cke-saved-href="/pkg/pstricks-base" href="/pkg/pstricks-base">pstricks</a>, it can produce either PostScript or PDF output.<br></p>]],
+  email = "pgf-tikz@tug.org",
+  license = "fdl;gpl2;lppl1.3c",
+  note = [[
+The release files are signed using a detached signature.  You can obtain the
+signature from the GitHub release page
 
-local git = {
-  tag = runcmd("git describe --abbrev=0 --tags"),
-  HEAD = runcmd("git rev-parse --abbrev-ref HEAD"),
+    https =//github.com/pgf-tikz/pgf/releases/download/${{ env.GIT_TAG }}/pgf_${{ env.GIT_TAG }}.ctan.flatdir.zip.sig]],
```

Comment:
@josephwright One last question: How do I get the `tagname` in here?

Reply:
At the moment there's not an easy way to do this that I can think of: will need some function you can call within the table.

Reply:
I saw there is `note_file` so I can just generate that during CI with the link from `upload-release-asset`.